crypto/src/secp256k1.rs

```diff
@@ -78,7 +79,7 @@ impl Verifier<Secp256k1Signature> for Secp256k1PublicKey {
 impl AsRef<[u8]> for Secp256k1PublicKey {
     fn as_ref(&self) -> &[u8] {
         self.bytes
-            .get_or_try_init::<_, eyre::Report>(|| Ok(self.pubkey.to_bytes()))
```

Comment:
qq: why not use .bytes directly (that we have from OnceCell)?

Reply:
this `bytes` OnceCell may not be populated, `get_or_try_init` tries to retrieve it by getting `bytes` directly, otherwise it will try populate it 

get_or_try_init:
```
Gets the contents of the cell, initializing it with f if the cell was empty. If the cell was empty and f failed, an error is returned.
```